packages/analytics-js-plugins/src/errorReporting/utils.ts

```diff
@@ -0,0 +1,192 @@
+    case ErrorType.UNHANDLEDEXCEPTION: {
+      const { error } = err as ErrorEvent;
+      return {
+        component: 'unhandledException handler',
+        tolerateNonErrors: true,
+        errorFramesToSkip: 1,
+        normalizedError: error || err,
+      };
+    }
+    case ErrorType.UNHANDLEDREJECTION: {
+      const error = err as PromiseRejectionEvent;
+      return {
+        component: 'unhandledrejection handler',
+        tolerateNonErrors: false,
+        errorFramesToSkip: 1,
+        normalizedError: error.reason,
+      };
+    }
+    case ErrorType.HANDLEDEXCEPTION:
+    default:
+      return {
+        component: 'notify()',
+        tolerateNonErrors: true,
+        errorFramesToSkip: 2,
+        normalizedError: err,
+      };
+  }
+};
+
+const createNewBreadcrumb = (message: string, metaData?: BreadcrumbMetaData): Breadcrumb => ({
+  type: 'manual',
+  name: message,
+  timestamp: new Date(),
+  metaData: metaData ?? {},
+});
+
+const getReleaseStage = () => {
+  const host = globalThis.location.hostname;
+  return host && DEV_HOSTS.includes(host) ? 'development' : '__RS_BUGSNAG_RELEASE_STAGE__';
+};
+
+const getAppStateForMetadata = (state: ApplicationState): Record<string, any> => {
+  const stateStr = json.stringifyWithoutCircular(state, false, APP_STATE_EXCLUDE_KEYS);
+  return stateStr !== null ? JSON.parse(stateStr) : {};
+};
+
+const getURLWithoutQueryString = () => {
+  const url = globalThis.location.href.split('?');
+  return url[0];
+};
+
+const getErrorContext = (event: any) => {
+  const { message } = event;
+  let context = message;
+
+  // Hack for easily grouping the script load errors
+  // on the dashboard
+  if (message.includes('Error in loading a third-party script')) {
+    context = 'Script load failures';
+  }
+  return context;
+};
+
+const getBugsnagErrorEvent = (
+  payload: ErrorFormat,
+  errorState: ErrorState,
+  state: ApplicationState,
+): ErrorEventPayload => ({
+  notifier: {
+    name: NOTIFIER_NAME,
+    version: '__PACKAGE_VERSION__',
+    url: SDK_GITHUB_URL,
+  },
+  events: [
+    {
+      payloadVersion: '5',
+      exceptions: clone(payload.errors),
+      severity: errorState.severity,
+      unhandled: errorState.unhandled,
+      severityReason: errorState.severityReason,
+      app: {
+        version: state.context.app.value.version,
+        releaseStage: getReleaseStage(),
+      },
+      device: {
+        locale: state.context.locale.value ?? undefined,
+        userAgent: state.context.userAgent.value || undefined,
+        time: new Date(),
+      },
+      request: {
+        url: getURLWithoutQueryString() as string,
+        clientIp: '[NOT COLLECTED]',
+      },
+      breadcrumbs: clone(state.reporting.breadcrumbs.value),
+      context: getErrorContext(payload.errors[0]),
+      metaData: {
+        sdk: {
+          name: 'JS',
+          installType: '__MODULE_TYPE__',
+        },
+        state: getAppStateForMetadata(state) ?? {},
+        source: {
```

Comment:
https://github.com/rudderlabs/rudder-sdk-js/blob/5dfd22c9bdf6368c00161c693a5957d782c320f3/packages/analytics-js-plugins/src/bugsnag/utils.ts#L73-L75

Reply:
It is already there.

Reply:
Only snippet version should be included.

Reply:
Done